openmp/runtime/test/atomic/omp-atomic-compare-signedness.c

```diff
@@ -11,7 +11,12 @@
 // UNSUPPORTED: gcc
 
 // High parallelism increases our chances of detecting a lack of atomicity.
+#ifdef __ve__
+// VE's pthread_create supports 64 threads at a maximum.
```

Comment:
If that's the case, then you want to set the limit (such as `__kmp_sys_max_nth`) in `kmp.h` as well.

Reply:
Thanks.  Modifying initial value for `__kmp_sys_max_nth` is not enough for our cases since `__kmp_runtime_initialize` is implemented assuming NPTL which assumes infinity number of threads.  I modify both `KMP_MAX_NTH` and the mechanism in `__kmp_runtime_initialize`.

Reply:
It would be better to split the patch into two: one is about test, as the previous version, and one for the changes in the header, etc.

Reply:
Will do.  Thanks.

Reply:
I created #66729 .  I'm not sure whether I create another patch for test or not.  Test itself passes after applying #66729 with following warnings.  I can say this is good regression tests for #66729 .  I also can say it's good to change test too for let ppl know this limitation.  Do you have any opinion?
```
OMP: Warning #96: Cannot form a team with 256 threads, using 64 instead.
OMP: Hint Consider unsetting KMP_DEVICE_THREAD_LIMIT (KMP_ALL_THREADS), KMP_TEAMS_THREAD_LIMIT, and OMP_THREAD_LIMIT (if any are set).
```

Reply:
Good to hear the test can pass after #66729. I'm more inclined to leave the test alone since we don't know what users can really do and the number looks totally reasonable from OpenMP's perspective. It is just our implementation can't handle it correctly.

Reply:
That make sense too.  I'll leave the test as is.  Thank you for your advice.